### PR TITLE
Move Dynamic Analysis Specs to Setup folder

### PR DESCRIPTION
**Pull request by toropok**
## Description of the issue/feature this PR addresses

Now it lives in bika_setup folder  

## Current behavior before PR

Now it lives in bika_setup folder 

## Desired behavior after PR is merged

move to Setup folder
--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

**Comment by toropok** [comment]
@ramonski please check it

File: src/senaite/core/upgrade/v02_06_000.py [hunk 1]
```diff
@@ -1641,6 +1641,42 @@ def migrate_attachmenttypes_to_dx(tool):
...
 
 
+@upgradestep(product, version)
+def move_dynamicanalysisspecs(tool):
+    """Move dynamic analysis specs to senaite package
+    """
+
+    # run required import steps
+    tool.runImportStepFromProfile(profile, "typeinfo")
+    tool.runImportStepFromProfile(profile, "workflow")
```
Comment (ramonski) [inline src/senaite/core/upgrade/v02_06_000.py]: Please remove

File: src/senaite/core/upgrade/v02_06_000.py [hunk 2]
```diff
@@ -1641,6 +1641,42 @@ def migrate_attachmenttypes_to_dx(tool):
...
+
+    # get the old container
+    origin = api.get_setup().get("dynamic_analysisspecs")
+    if not origin:
+        # old container is already gone
+        return
+
+    # get the destination container
+    destination = get_setup_folder("dynamic_analysisspecs")
```
Comment (ramonski) [inline src/senaite/core/upgrade/v02_06_000.py]: Can you change the folder name to `dynamicanalysisspecs` to match the naming convention of the other DX setup folders?

Comment (toropok) [reply src/senaite/core/upgrade/v02_06_000.py]: yes, indeed, noticed about convention

File: src/senaite/core/upgrade/v02_06_000.py (hunk 1 above)
Comment (toropok) [reply src/senaite/core/upgrade/v02_06_000.py]: yes, Commander. We've disputed is is required or not, and I left it there

File: src/bika/lims/content/dynamic_analysisspec.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ class IDynamicAnalysisSpec(model.Schema):
         required=True)
 
     @invariant
-    def validate_sepecs_file(data):
+    def validate_specs_file(data):
```
Comment (ramonski) [inline src/bika/lims/content/dynamic_analysisspec.py:54]: good catch!

**Comment by ramonski** [comment]
Hmm, I think we have a problem here (and probably also with the other "move" migrators).

Currently, the listing view is not displayed for the new `dynamicanalysisspecs` folder:

Dynamic Analysisspecs

The reason is, that the object still implements the interface from `bika.lims`:

Dynamic Analysis Specifications 2024-06-19 10 PM-23-32

Which is caused by the old FTI (file type information) stored in `portal_types`:

Dynamic Analysis Specifications 2024-06-19 10 PM-27-15

So I guess there are two solutions to this:

1. import the `typeinfo` step, but ensure old AT types are flushed, e.g.:

```
    # ensure old AT types are flushed first
    remove_at_portal_types(tool)

    # run required import steps
    tool.runImportStepFromProfile(profile, "typeinfo")
```

2. change the FTI interface directly.


Remember: The reason why we removed the import steps was because of this traceback: https://github.com/senaite/senaite.core/pull/2554#issuecomment-2175762913

Therefore, I would add the code above and add the same to the preceeding "move" migrators.

File: src/bika/lims/content/dynamic_analysisspec.py (hunk 3 above)
Comment (toropok) [reply src/bika/lims/content/dynamic_analysisspec.py:54]: @ramonski 

great! revert back migration step and add to previous "move" methods.

migration flow seems still not very clear sometimes :) our test instance updates incrementally and we missed some errors you found.

Comment (ramonski) [reply src/bika/lims/content/dynamic_analysisspec.py:54]: > migration flow seems still not very clear sometimes

Yes, I can completely understand that!

The fact that the configuration lives in the DB and the interfaces/classes are pickled in the objects, make the migration really painful.
But now you understand better, why the whole migration process can not be done fast and is also difficult to estimate the needed efforts ...

**Review by ramonski — APPROVED** [review]
Thanks for your work Leonid!

**Pull request merged**